ZionEscape/Scene.h

```diff
@@ -0,0 +1,155 @@
+#pragma once
+
+#include "SceneSpawner.h"
+
+using namespace System::Drawing;
+using namespace System::Collections::Generic;
+
+ref class Scene {
+  Bitmap^ image;
+  bool up, down, right, left;
+  List<SceneSpawner^>^ spawners;
+  Rectangle drawingArea;
+public:
+  Scene(bool up, bool down, bool right, bool left, Point pos) {
+    this->spawners = gcnew List<SceneSpawner^>;
+    this->up = up;
+    this->down = down;
+    this->right = right;
+    this->left = left;
+    this->ImageSelector();
+    this->drawingArea = Rectangle(pos.X, pos.Y, this->image->Width, this->image->Height);
```

Comment:
Podrías usar el constructor de [Rectangle](https://docs.microsoft.com/en-us/dotnet/api/system.drawing.rectangle.-ctor?view=netframework-4.8#System_Drawing_Rectangle__ctor_System_Drawing_Point_System_Drawing_Size_) que toma un `Point` y un `Size`.

```suggestion
    this->drawingArea = Rectangle(pos, this->image->Size);
```

Reply:
Hola, gracias por las mejoras mencionadas! Respecto a lo de añadir los *header guards*, ¿no cumpliría la misma función que el `#pragma once`?

Reply:
`#pragma once` es una directiva incluida en compiladores recientes que hace eso mismo. Aun así, he decidido añadir los _header guards_ a todos los archivos debido a unas dependencias circulares que se estaban generando.